packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -1651,6 +1661,110 @@ export class AiAgentService {
         });
     }
 
+    async setArtifactVersionVerified(
+        user: SessionUser,
+        {
+            agentUuid,
+            artifactUuid,
+            versionUuid,
+            verified,
+        }: {
+            agentUuid: string;
+            artifactUuid: string;
+            versionUuid: string;
+            verified: boolean;
+        },
+    ): Promise<void> {
+        const { organizationUuid } = user;
+        if (!organizationUuid)
+            throw new ForbiddenError(`Organization not found`);
+
+        const isCopilotEnabled = await this.getIsCopilotEnabled(user);
+        if (!isCopilotEnabled)
+            throw new ForbiddenError(`Copilot is not enabled`);
+
+        const agent = await this.aiAgentModel.getAgent({
+            organizationUuid,
+            agentUuid,
+        });
+
+        if (!agent) {
+            throw new NotFoundError(`Agent not found: ${agentUuid}`);
+        }
+
+        const artifact = await this.aiAgentModel.getArtifact(
+            artifactUuid,
+            versionUuid,
+        );
+        if (!artifact) {
+            throw new NotFoundError(
+                `Artifact version not found: ${artifactUuid}/${versionUuid}`,
+            );
+        }
+
+        const thread = await this.aiAgentModel.getThread({
+            organizationUuid,
+            agentUuid,
+            threadUuid: artifact.threadUuid,
+        });
+        if (!thread) {
+            throw new NotFoundError(`Thread not found: ${artifact.threadUuid}`);
+        }
+
+        const hasAccess = await this.checkAgentThreadAccess(
+            user,
+            agent,
+            thread.user.uuid,
+        );
+        if (!hasAccess) {
+            throw new ForbiddenError(
+                'Insufficient permissions to modify this artifact',
+            );
+        }
+
+        await this.aiAgentModel.setArtifactVersionVerified(
+            versionUuid,
+            verified,
+            user.userUuid,
+        );
+    }
+
+    async embedArtifactVersion(payload: {
+        artifactVersionUuid: string;
+        title: string | null;
+        description: string | null;
+    }): Promise<void> {
+        try {
+            const text = [payload.title, payload.description]
+                .filter(Boolean)
+                .join(' ');
```

Comment:
you can do something like so they are separated and better matched
title: ${title}
description: ${description}